p2p/peering.go

```diff
@@ -1174,11 +1170,9 @@ func (rp *RemotePeer) receivedHeaders(ctx context.Context, msg *wire.MsgHeaders)
 
 	// Track the height of the last received header.
 	if prevHeight > 0 {
-		rp.lastHeightMu.Lock()
-		if int32(prevHeight) > rp.lastHeight {
-			rp.lastHeight = int32(prevHeight)
+		if int32(prevHeight) > rp.lastHeight.Load() {
```

Comment:
This changes the semantics.  Previously, it was impossible for it to go backwards, but that is no longer true with this change.  That may or may not be a problem.  It would take spending more time to dig into the surrounding code to say for sure, but typically going backwards with heights can cause issues.

Reply:
For reference, this can now go backwards as follows:

1) The current `rp.lastHeight=1000`
2) The current goroutine is running with `prevHeight=1001`
3) Some other goroutine is running with `prevHeight=1002`
4) Both goroutines load the atomic simultaneously thereby causing the condition to be true
5) The other goroutine stores 1002
6) The current goroutine stores 1001, going backwards and losing the 1002 update altogether